crates/sui-core/src/authority.rs

```diff
@@ -2583,9 +2589,18 @@ impl AuthorityState {
 
     async fn reopen_epoch_db(&self, new_committee: Committee) {
         info!(new_epoch = ?new_committee.epoch, "re-opening AuthorityEpochTables for new epoch");
+        let new_epoch = new_committee.epoch;
+        let epoch_number_to_retain = self.epoch_number_to_retain;
+
         let epoch_tables = self
             .epoch_store()
             .new_at_next_epoch(self.name, new_committee);
+        let parent_path = epoch_tables.get_parent_path();
+        spawn_monitored_task!(prune_old_epoch_data(
```

Comment:
@phoenix-o correct me if I am reading it wrong, but it seems like we are creating a new task for every new epoch. Would it make more sense to have just one task running in the bg doing gc periodically? I think it might be better because we will avoid race condition with multiple such tasks running concurrently with shorter duration epochs like in tests (where we create an epoch every 5 minutes). Also, if db drop fails - we can keep retrying in the background

Reply:
changed the flow to have a single pruner task